src/rolling/directory.rs

```diff
@@ -140,7 +140,7 @@ impl RollingReader {
     ///
     /// If no block was read, positions itself at the beginning.
     pub async fn into_writer(mut self) -> io::Result<RollingWriter> {
-        let offset = self.block_id * crate::BLOCK_NUM_BYTES;
+        let offset = (self.block_id - 1) * crate::BLOCK_NUM_BYTES;
```

Comment:
I'm scared here.

a) Aren't you overriding data from the last block with this?
b) we are sure that  `self.block_id > 1`? Can we add an assert on it or even make `block_id` a NonZero if it is possible?

Reply:
>  b) we are sure that self.block_id > 1? Can we add an assert on it or even make block_id a NonZero if it is possible?

as per comment on the `block_id` field, `block_id` is the "number of the next block to read". Currently, short of an overflow, there is no way for block_id to be 0 (but there will be by making the changes suggested below).
I'm not sure what prompted to have block_id be the number of the *next* block, I think it should be made into the number of the current block, which would remove that -1 and always initialize it at 0 instead of 1.

> a) Aren't you overriding data from the last block with this?

I'm not, but for non-local reasons. When a FrameReader info_writer() is called, it keeps its cursor. In the following scenario:
```
write_frame([1,2,3,4,5,6])
reload()
write_frame([5,6,7,8])
```
the previous behaviour was to store (assuming 8 bytes blocks for conciseness, and assuming a frame and its header can fit in 4 bytes for the same reason):
```
bl1: 01 02 03 04 00 00 00 00
bl2: 00 00 00 00 05 06 07 08
```
When trying to read again, the reader would read the first frame, and find a header full of zero, so it assumes end of data, and misses the second write.

With the new behavior the result is
```
bl1: 01 02 03 04 05 06 07 08
bl2: 00 00 00 00 00 00 00 00
```
The same block get written in, but the write start at the first position where there is an all-zero frame header

Note that this is the behaviour the doc comment hints toward: 
> Creates a write positioned at the beginning of the last read block.